rules/0131.go

```diff
@@ -0,0 +1,185 @@
+					return
+				}
+
+				// Rule check: Establish that for methods such as `GetFoo`, the request
+				// message is named `GetFooRequest`.
+				requestMessageName := string(m.Input().Name())
+				if requestMessageName != methodName+"Request" {
+					problems = append(problems, lint.Problem{
+						Message: fmt.Sprintf(
+							"Get RPCs should have a request message named after the RPC, such as %q.",
+							methodName+"Request",
+						),
+						Suggestion: methodName + "Request",
+						Descriptor: m,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// The Get standard method should only have expected fields.
+func checkRequestMessageNameField() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0131", "request_message", "name_field"),
+			Description:  "check that a name field is present",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get- methods for the purpose of this rule;
+				// ignore everything else.
+				if !strings.HasPrefix(string(m.Name()), "Get") {
+					return
+				}
+
+				// Rule check: Establish that a name field is present.
+				nameField := m.Input().Fields().ByName("name")
+				if nameField == nil {
+					problems = append(problems, lint.Problem{
+						Message:    fmt.Sprintf("method %q has no `name` field", m.Name()),
+						Descriptor: m.Input(),
+					})
+					return problems, nil
+				}
+
+				// Rule check: Establish that the name field is a string.
+				if nameField.Kind() != p.StringKind {
+					problems = append(problems, lint.Problem{
+						Message:    "`name` field on Get RPCs should be a string",
+						Descriptor: nameField,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Get methods should not have unrecognized fields.
+func checkRequestMessageUnknownFields() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0131", "request_message", "unknown_fields"),
+			Description:  "check that there are no unknown fields",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get- methods for the purpose of this rule;
+				// ignore everything else.
+				if !strings.HasPrefix(string(m.Name()), "Get") {
+					return
+				}
+
+				// Rule check: Establish that there are no other fields besides `name`.
+				fields := m.Input().Fields()
+				for i := 0; i < fields.Len(); i++ {
+					field := fields.Get(i)
+					if string(field.Name()) != "name" {
```

Comment:
Some fields are accepted, e.g. show_deleted, or a view field.

Reply:
Oooh I forgot about views and read masks, thanks.
Is `show_deleted` valid for Get? I thought it was just List. (Honestly asking.)

Reply:
Added everything except `show_deleted`, which I _think_ does not belong in Gets. (Totally willing to add it if needed.)